jetstream/tests/integration/conftest.py

```diff
@@ -1,8 +1,90 @@
+from google.cloud import bigquery
+from google.api_core.exceptions import NotFound
+from pathlib import Path
 import pytest
+import random
+import string
+
+from jetstream.analysis import BigQueryClient
+
+TEST_DIR = Path(__file__).parent.parent
 
 
 def pytest_runtest_setup(item):
     if "FLAKE8" in item.nodeid or "BLACK" in item.nodeid:
         return
     if not item.config.getoption("--integration", False):
         pytest.skip("Skipping integration test")
+
+
+@pytest.fixture
+def project_id():
+    """Provide a BigQuery project ID."""
+    return "jetstream-integration-test"
+
+
+@pytest.fixture
+def temporary_dataset(request):
+    """Fixture for creating a random temporary BigQuery dataset."""
+    # generate a random test dataset to avoid conflicts when running tests in parallel
+    test_dataset = "test_" + "".join(random.choice(string.ascii_lowercase) for i in range(12))
+
+    project_id = request.getfixturevalue("project_id")
+    client = bigquery.Client(project_id)
+    client.create_dataset(test_dataset)
+
+    yield test_dataset
+
+    # cleanup and remove temporary dataset
+    client.delete_dataset(test_dataset, delete_contents=True, not_found_ok=True)
+
+
+@pytest.fixture
+def client(request):
+    """Provide a BigQuery client."""
+    project_id = request.getfixturevalue("project_id")
+    temporary_dataset = request.getfixturevalue("temporary_dataset")
+    return BigQueryClient(project_id, temporary_dataset)
```

Comment:
The dependency on temporary_dataset here is a little surprising to me and requires some reasoning about fixture lifetimes; if a function asks for both `client` and `temporary_dataset`, is the client instantiated with the same dataset that the test receives?

Is the client fixture adding much?

Reply:
Just checked, the client is instantiated with the same dataset that the test receives